pandas/core/groupby.py

```diff
@@ -1883,6 +1883,7 @@ def __init__(self, index, grouper=None, obj=None, name=None, level=None,
         # pre-computed
         self._was_factor = False
         self._should_compress = True
+        self._is_categorical = False
```

Comment:
hmm, this basically duplicates ``_was_factor``. I need you to disambiguate when that is being set incorrectly.